stl/error_handler.py

```diff
@@ -14,7 +14,71 @@
 
 """Module for handling errors from parsing an STL file."""
 
-import error_formatter
+import logging
+
+import stl.error_formatter
+
+
+# Low level parse errors live in the 100s
+_ERROR_ID_UNKNOWN_PARSE_ERROR = 100
+_ERROR_MSG_UNKNOWN_PARSE_ERROR = 'There was an unknown parse error :('
+
+_ERROR_ID_MISSING_SEMICOLON = 101
+_ERROR_MSG_MISSING_SEMICOLON = 'Missing semicolon.'
+
+_ERROR_ID_MISSING_CLOSING_CURLY_BRACE = 102
+_ERROR_MSG_MISSING_CLOSING_CURLY_BRACE = 'Missing closing curly brace }.'
+
+# Higer level parse errors live in the 200s
+_ERROR_ID_UNKNOWN_STATE_VALUE = 201
+_ERROR_MSG_UNKNOWN_STATE_VALUE = 'Unknown state value. Did you mean {}?'
+
+_ERROR_ID_MISSING_POST_STATES = 202
+_ERROR_MSG_MISSING_POST_STATES = ('Transitions require "post_states". If there'
+                                  ' are no explicit post_states use an empty'
+                                  ' list (post_states = []).')
+
+_ERROR_ID_MISSING_PRE_STATES = 203
+_ERROR_MSG_MISSING_PRE_STATES = ('Transitions require "pre_states", and the'
+                                 ' "pre_states" must be non-empty.')
+
+_ERROR_ID_EMPTY_PRE_STATES = 204
+_ERROR_MSG_EMPTY_PRE_STATES = ('Transitions require non-empty "pre_states".')
+
+# Lex erros live in the 300s
+_ERROR_ID_UNKNOWN_LEX_ERROR = 300
+_ERROR_MSG_UNKNOWN_LEX_ERROR = 'There was an unknown lexing/tokenizing error :('
+
+_ERROR_ID_MISSING_STRING_QUOTE = 301
+_ERROR_MSG_MISSING_STRING_QUOTE = 'Missing " " on string literal.'
+
+_ERROR_ID_SINGLE_QUOTE = 302
+_ERROR_MSG_SINGLE_QUOTE = 'Use double quotes (" ") for string literals.'
```

Comment:
Better to define a class, Error?

Reply:
Done. I'll split the class out into another file also to keep things clean; I don't have access to proper editor right now though, and splitting across files is not convenient in the github web ui >.>